src/sentry/sentry_apps/tasks/sentry_apps.py

```diff
@@ -214,73 +217,88 @@ def _process_resource_change(
     retryer: Task | None = None,
     **kwargs: Any,
 ) -> None:
-    # The class is serialized as a string when enqueueing the class.
-    model: type[Event] | type[Model] = TYPES[sender]
-    instance: Event | Model | None = None
-
-    project_id: int | None = kwargs.get("project_id", None)
-    group_id: int | None = kwargs.get("group_id", None)
-    if sender == "Error" and project_id and group_id:
-        # Read event from nodestore as Events are heavy in task messages.
-        nodedata = nodestore.backend.get(Event.generate_node_id(project_id, str(instance_id)))
-        if not nodedata:
-            extra = {"sender": sender, "action": action, "event_id": instance_id}
-            logger.info("process_resource_change.event_missing_event", extra=extra)
+    with SentryAppInteractionEvent(
+        operation_type=SentryAppInteractionType.PREPARE_WEBHOOK,
+        event_type=f"process_resource_change.{sender}_{action}",
```

Comment:
We dont have an 'event' string at this point so making something up

Reply:
will this be pulled from `kwargs` later on? what will this represent?

Reply:
wdym pulled from `kwargs`, like added to logging? My goal for this `event_type` is that we can point to some process and function and be like we're failing here. So in that way, this `event_type` represents the location/function.

Additionally we do add `event_type` to our logs and as a metric tag

Reply:
is there a hard coded list of `event_types`? is the `event_type` constructed from the incoming parameters?

Reply:
changed this btw now we construct the event prior to starting the lifecycle method and if the event doesnt fit nicely into our enums throw errors